software/base/src/main/java/org/apache/brooklyn/entity/machine/MachineInitTasks.java

```diff
@@ -75,13 +72,19 @@ public Void call() {
     /**
      * Returns a queued {@link Task} which stops iptables on the given machine.
      */
-    public Task<Void> stopIptablesAsync(final SshMachineLocation machine) {
-        return DynamicTasks.queue("stop iptables", new Callable<Void>() {
-            public Void call() {
-                stopIptablesImpl(machine);
-                return null;
-            }
-        });
+    public ProcessTaskWrapper<Integer> stopIptablesAsync(final SshMachineLocation machine) {
+
+        log.info("Stopping iptables for {} at {}", entity(), machine);
+
+        List<String> cmds = ImmutableList.<String>of();
+        if (isLocationFirewalldEnabled(machine)) {
+            cmds = ImmutableList.of(IptablesCommands.firewalldServiceStop(), IptablesCommands.firewalldServiceStatus());
+        } else {
+            cmds = ImmutableList.of(IptablesCommands.iptablesServiceStop(), IptablesCommands.iptablesServiceStatus());
+        }
+        ProcessTaskFactory<Integer>  stopIptables = SshTasks.newSshExecTaskFactory(machine, cmds.toArray(new String[cmds.size()]))
+                .summary("stop iptables");
+        return DynamicTasks.queue(stopIptables);
```

Comment:
The above 3 lines repeat multiple times in the file, in functions `insertIptablesRulesOnCommandBatches`, `serviceIptablesSave`, `listIptablesRules`. Better create a single helper function which takes the machine, list of commands and summary for the task instead of creating a dedicated function for each call.